source_modelling/srf.py

```diff
@@ -0,0 +1,407 @@
+        points = pd.DataFrame(
+            (read_srf_point(srf_file_handle) for _ in range(point_count))
+        )
+        return SrfFile(version, headers, points)
+
+
+def write_srf_header(srf_file: TextIO, header: pd.DataFrame) -> None:
+    """Write a list of SRF plane segments as a header.
+
+    Parameters
+    ----------
+    srf_file : TextIO
+        The SRF file to write to.
+    header : pd.DataFrame
+        The list of SrfSegments to write.
+    """
+    srf_file.write(f"PLANE {len(header)}\n")
+    srf_file.write(
+        header.to_string(
+            index=False,
+            header=None,
+            formatters={
+                "elon": lambda elon: f"{elon:.6f}",
+                "elat": lambda elat: f"{elat:.6f}",
+                "len": lambda len: f"{len:.4f}",
+                "wid": lambda wid: f"{wid:.4f}",
+            },
+        )
+        + "\n"
+    )
+
+
+def write_point_count(srf_file: TextIO, point_count: int) -> None:
+    """Write out a POINTS declaration for the number of points in an SRF file..
+
+    Parameters
+    ----------
+    srf_file : TextIO
+        The SRF file to write to.
+    point_count : int
+        The number of points to write.
+    """
+    srf_file.write(f"POINTS {point_count}\n")
+
+
+def write_slip(srf_file: TextIO, slips: np.ndarray) -> None:
+    """Write out slip values to an SRF file.
+
+    Parameters
+    ----------
+    srf_file : TextIO
+        The SRF file to write to.
+    slips : np.ndarray
+        The slip values to write.
+    """
+    for i in range(0, len(slips), 6):
+        srf_file.write(
+            "  "
+            + "  ".join(f"{slips[j]:.5E}" for j in range(i, min(len(slips), i + 6)))
+            + "\n"
+        )
+
+
+def write_srf_point(srf_file: TextIO, point: pd.Series) -> None:
+    """Write out a single SRF point.
+
+    Parameters
+    ----------
+    srf_file : TextIO
+        The SRF file to write to.
+    point : pd.Series
+        The point to write.
+    """
+    srf_file.write(
+        f"{point['lon']:.6f} {point['lat']:.6f} {point['dep']:g} {point['stk']:g} {point['dip']:g} {point['area']:.4E} {point['tinit']:.4f} {point['dt']:.6E}\n"
+    )
+    srf_file.write(
+        f"{point['rake']:g} {point['slip1']:.4f} {len(point['slipt1'])} {point['slip2']:.4f} {len(point['slipt2'])} {point['slip3']:.4f} {len(point['slipt3'])}\n"
+    )
+    if len(point["slipt1"]):
+        write_slip(srf_file, point["slipt1"])
+    if len(point["slipt2"]):
+        write_slip(srf_file, point["slipt2"])
+    if len(point["slipt3"]):
+        write_slip(srf_file, point["slipt3"])
+
+
+def write_srf(srf_ffp: Path, srf: SrfFile) -> None:
+    """Write an SRF object to a filepath.
+
+    Parameters
+    ----------
+    srf_ffp : Path
+        The filepath to write the srf object to.
+    srf : SrfFile
+        The SRF object.
+    """
+    with open(srf_ffp, mode="w", encoding="utf-8") as srf_file_handle:
+        write_version(srf_file_handle)
```

Comment:
Same comment as above.

Reply:
In my opinion this module should only have 2 or maybe 4 functions, unless there is a good reason for splitting it up this much?

Reply:
It's a legacy of a slightly older design. I can easily change that.

Reply:
Cool, I think that would be a good idea.